dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/debug/UiRumDebugListener.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.debug
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.graphics.Color
+import android.os.Bundle
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.annotation.UiThread
+import androidx.core.graphics.blue
+import androidx.core.graphics.green
+import androidx.core.graphics.red
+import androidx.core.view.setPadding
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.monitor.AdvancedRumMonitor
+import java.util.Locale
+import kotlin.math.pow
+
+internal class UiRumDebugListener(
+    private val rumMonitorProvider: () -> AdvancedRumMonitor? = {
+        GlobalRum.get() as? AdvancedRumMonitor
+    }
+) :
+    Application.ActivityLifecycleCallbacks, RumDebugListener {
+
+    internal var rumViewsContainer: LinearLayout? = null
+    internal var rumMonitor: AdvancedRumMonitor? = null
+
+    private val viewsSnapshot = mutableListOf<String>()
+
+    // region Application.ActivityLifecycleCallbacks
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        // no-op
+    }
+
+    override fun onActivityStarted(activity: Activity) {
+        // no-op
+    }
+
+    override fun onActivityResumed(activity: Activity) {
+
+        val contentView = findContentView(activity)
+        if (contentView == null) {
+            devLogger.w(CANNOT_FIND_CONTENT_VIEW_MESSAGE)
+            return
+        }
+
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        val rumMonitor = rumMonitorProvider()
```

Comment:
Why do you need a lambda here? In theory the RumMonitor is set once and for all (we only have a public `registerIfAbsent`, so once it's set it cannot change). Maybe having a `lazy` property would work better?

Reply:
I was having in mind in mind a case where `RumMonitor` can be removed after it is added (if we allow such with SDK v2), but I think I will reject idea of using provider.